### DISCUSSION TITLE
custom gitlab server - Cannot determine forge repository

### DISCUSSION THREAD

**Discussion by yashasolutions** [magit/forge#356] [Support] [answered]
Hi,

I am struggling to get emacs forge to recognize custom gitlab server.
I have added to my config

```
(with-eval-after-load 'magit
  (require 'forge))

(with-eval-after-load 'forge
  (add-to-list 'forge-alist
               '("glcustom"
                 "gitlab.custom.site/api/v4"
                 "gitlab.custom.site"
                 forge-gitlab-repository)))
```

Yet I keep getting when I run M-x forge-pull or M-x forge-add-repository
```
Cannot determine forge repository. git@gitlab.custom.site:group/project.git isn't a project url
```
I have I checked the content of forge-alist, it's in there.

I have added the following to the .ssh/config the 

```yaml
Host glcustom
    User git
    Hostname gitlab.custom.site
```

Can't seem to see what's wrong...

**Comment by tarsius** [comment, marked as answer]
`forge-alist` entries should have the form `(GITHOST APIHOST ID CLASS)` but you went with `(ID APIHOST GITHOST CLASS)`.

**Comment by yashasolutions** [comment]
perfect! worked! Thanks!
